pkg/common/resources/resources.go

```diff
@@ -44,6 +46,52 @@ func (q Quantity) string() string {
 	return strconv.FormatInt(int64(q), 10)
 }
 
+// Util struct to keep track of application resource usage
+type UsedResourceTracker struct {
+	// Two level map for aggregated resource usage
+	// With instance type being the top level key, the mapped value is a map:
+	//   resource type (CPU, memory etc) -> the aggregated used time (in seconds) of the resource type
+	//
+	UsedResourceMap map[string]map[string]int64
+
+	sync.RWMutex
+}
+
+// Aggregate the resource usage to UsedResourceMap[instType]
+// The time the given resource used is the delta between the resource createTime and currentTime
+func (urt *UsedResourceTracker) AggregateUsedResource(instType string,
+	resource *Resource, createTime time.Time) {
+	urt.Lock()
+	defer urt.Unlock()
+
+	releaseTime := time.Now()
+	timeDiff := int64(releaseTime.Sub(createTime).Seconds())
+	aggregatedResourceTime, ok := urt.UsedResourceMap[instType]
+	if !ok {
+		aggregatedResourceTime = map[string]int64{}
+	}
+	for key, element := range resource.Resources {
+		resourceSecondsKey := key + "Seconds"
```

Comment:
Thanks for your detailed review. 

What custom resource types you were referring too? are they not going to be based on seconds? 

To explain why I did this:

I'm trying to make it easy for viewer to grasp the meaning of the numbers reported when one looks at the app summary reported.  

Plus, the measurement of resource usage should be "resourceAmount * timeLength", not just "resourceAmount". Without a time unit ("Seconds" here) in the name, one can misread it as amount of resource, instead of "amount of resource * timeLength". If you notice, current yunikorn UI does report memory/vcore as "resources used", which is a misnomer in the UI, because what the UI displayed is a snapshot of currently used resourceAmount, not "resourceAmount * timeLength". 

In addition, having the "Seconds" string in it makes it easy to adapt to since we are migrating from YARN to yunikorn, as YARN does report memorySeconds and vcoreSeconds too. 

Wonder with the above explanation, would you still suggest removing the "Seconds"? 

Thanks.

Reply:
Keep the names consistent as this way a downstream user doesn't need to subtract off "Seconds" to get the original resource type. It's simply unnecessary, and adds additional complexity.